### PR TITLE
AEA-294 AEA-315 Improve existing demo guides.

### PR DESCRIPTION
**Pull request by Totoual**
## Proposed changes

Part2 of the improvement of the existing demos. 

## Fixes

 -

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/car-park-skills.md [hunk 1]
```diff
@@ -1,16 +1,21 @@
-The full Fetch.ai car park AEA demo is documented in its own repo [here](https://github.com/fetchai/carpark_agent).
-
-This demo allows you to test the AEA functionality of the car park AEA demo without the detection logic.
-
-It demonstrates how the AEAs trade car park information.
+The AEA car-park skills demonstrate an interaction between two AEAs.
 
+* The carpark_detection AEA provides data.
```
Comment (DavidMinarsch) [inline docs/car-park-skills.md]: provides information on the number of car parking spaces available in a given vicinity.

File: docs/car-park-skills.md [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
+* The carpark_client AEA that buys these data.
```
Comment (DavidMinarsch) [inline docs/car-park-skills.md]: AEA is interested in purchasing information on available car parking spaces in the same vicinity.

File: docs/car-park-skills.md [hunk 3]
```diff
@@ -34,13 +38,6 @@ aea add skill fetchai/carpark_detection:0.1.0
 aea install
 ```
 
-Second, add the ledger info to the aea config (`car_detector/aea-config.yaml`):
```
Comment (DavidMinarsch) [inline docs/car-park-skills.md]: This needs to stay..

File: docs/car-park-skills.md [hunk 4]
```diff
@@ -59,54 +56,121 @@ aea add skill fetchai/carpark_client:0.1.0
 aea install
 ```
 
-Second, add the ledger info to the aea config (`car_data_buyer/aea-config.yaml`):
```
Comment (DavidMinarsch) [inline docs/car-park-skills.md]: This needs to stay

File: docs/car-park-skills.md [hunk 5]
```diff
@@ -59,54 +56,121 @@ aea add skill fetchai/carpark_client:0.1.0
...
 ```
 
-To fund the car data buyer AEA on test-net you need to request some test tokens from a faucet.
+To generate and add a key for Ethereum use:
+```bash
+aea generate-key ethereum
+aea add-key ethereum eth_private_key.txt
+```
 
-First, print the address:
```
Comment (DavidMinarsch) [inline docs/car-park-skills.md]: The update configs step is only relevant when the agent was not fetched.

File: docs/gym-skill.md [hunk 6]
```diff
@@ -7,6 +7,14 @@ The AEA gym skill demonstrates how a custom Reinforcement Learning agent, that u
 
 Follow the <a href="../quickstart/#preliminaries">Preliminaries</a> and <a href="../quickstart/#installation">Installation</a> sections from the AEA quick start.
 
+##Discussion
+
+The `gym` example demonstrates the AEA framework's flexibility with respect to Reinforcement Learning using OpenAI's `gym` framework.
```
Comment (DavidMinarsch) [inline docs/gym-skill.md]: The `gym` skills demonstrate how to wrap a Reinforcement Learning agent in a skill.

File: docs/gym-skill.md [hunk 7]
```diff
@@ -7,6 +7,14 @@ The AEA gym skill demonstrates how a custom Reinforcement Learning agent, that u
...
+
+There is no immediate use case for this example as you can train an RL agent without the AEA proxy layer just fine (and faster).
```
Comment (DavidMinarsch) [inline docs/gym-skill.md]: remove this and next line

File: docs/ml-skills.md [hunk 8]
```diff
@@ -14,7 +14,7 @@ Follow the <a href="../quickstart/#preliminaries">Preliminaries</a> and <a href=
 ##Discussion
 
 The scope of the specific demo is to demonstrate how to create a simple AEA with integration of machine learning, and the usage of the AEA framework. The ml_data_provider AEA
-will provide some sample data and will deliver to the client upon payment. Once the client gets the data, it will train the model. The process can be found in the task.py file.
+will provide some sample data and will deliver to the client upon payment. Once the client gets the data, it will train the model. The process can be found in the tasks.py file.
```
Comment (DavidMinarsch) [inline docs/ml-skills.md]: `tasks.py` file

File: docs/thermometer-skills.md [hunk 9]
```diff
@@ -56,141 +56,112 @@ aea add skill fetchai/thermometer_client:0.1.0
...
+  ethereum:
+    address: https://ropsten.infura.io/v3/f00f7b3ba0e848ddbdc8941c527447fe
+    chain_id: 3
+    gas_price: 50
 ```
-## Demo instructions 2: Ethereum ledger payment
 
-A demo to run the same scenario but with a true ledger transaction on the Ethereum Ropsten `testnet`. 
-This demo assumes the thermometer client trusts the thermometer AEA to send the data upon successful payment.
+### Fund the weather client AEA
```
Comment (DavidMinarsch) [inline docs/thermometer-skills.md]: thermometer

File: docs/car-park-skills.md (hunk 5 above)
Comment (Totoual) [reply docs/car-park-skills.md]: It is sections. It is all about creating the AEAs. Then you have to update the configs based on the network you want to connect to. Even if you fetch the AEA it is going to be connected to the default network. The user must know how to change the network, from my point of view at least.

**Pull request merged**